account-kit/java/signer-sdk/src/main/java/com/alchemy/aa/Stamper.java

```diff
@@ -0,0 +1,209 @@
+package com.alchemy.aa;
+
+import com.alchemy.aa.core.TEKManager;
+import com.alchemy.aa.core.exceptions.NoInjectedBundleException;
+import com.alchemy.aa.core.exceptions.StamperNotInitializedException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.google.common.primitives.Bytes;
+import com.google.crypto.tink.subtle.Base64;
+import com.google.crypto.tink.subtle.EllipticCurves;
+
+import com.google.crypto.tink.subtle.Hex;
+import com.google.protobuf.InvalidProtocolBufferException;
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+import java.security.KeyFactory;
+import java.security.Security;
+import java.security.interfaces.ECPrivateKey;
+import java.security.interfaces.ECPublicKey;
+import java.security.Signature;
+import java.security.spec.ECPoint;
+import org.bitcoinj.base.Base58;
+import org.bouncycastle.jce.ECNamedCurveTable;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.jce.spec.ECPublicKeySpec;
+
+public class Stamper {
+
+    public record Stamp(String stampHeaderName, String stampHeaderValue) {
+    }
+
+    public record APIStamp(String publicKey, String scheme, String signature) {
+    }
+
+    public static String toJson(APIStamp apiStamp) throws JsonProcessingException {
+        ObjectWriter ow = new ObjectMapper().writer().withDefaultPrettyPrinter();
+        return ow.writeValueAsString(apiStamp);
+    }
+
+    public Stamper(TEKManager tekManager) {
+        if (Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) == null) {
+            Security.addProvider(new BouncyCastleProvider());
+        }
+
+        this.tekManager = tekManager;
+        this.bundlePublicKey = null;
+        this.bundlePrivateKey = null;
+    }
+
+    public Stamper() {
+        this(/* tekManager= */null);
+    }
```

Comment:
is this a construction flow we want to enable? you kinda always need a TEK manager to make this work.

I think we should have the following in addition to the one above:

```
public Stamper(targetPrivateKeyHex: String) {}

public Stamper(targetPrivateKey: byte[]) {}

public Stamper(targetKeySet: KeysetHandle) {}

// maybe these too?
public Stamper(targetPrivateKeyHex: String, bundle: String) {}

public Stamper(targetPrivateKey: byte[], bundle: String) {}

public Stamper(targetKeySet: KeysetHandle, bundle: String) {}
```